src/windows/views/emojis_listview.py

```diff
@@ -109,48 +106,49 @@ def add_file(self, filepath):
             # Log exception
             log.warning("Failed to import file: {}".format(str(ex)))
 
+    @pyqtSlot(int)
+    def group_changed(self, index=-1):
+        log.info("Group set to: {}".format(index))
+        item = get_app().window.emojiFilterGroup.itemData(index)
+        log.info("Filtering on {}".format(item))
+        self.group_model.setFilterFixedString(item)
+        self.group_model.setFilterKeyColumn(1)
+
+        self.refresh_view()
+
     @pyqtSlot(str)
-    def filter_changed(self):
+    def filter_changed(self, filter_text=None):
+        """Filter emoji with proxy class"""
+        log.info("Filter text: {}".format(filter_text or ""))
+        self.model.setFilterRegExp(QRegExp(filter_text, Qt.CaseInsensitive))
+        self.model.setFilterKeyColumn(0)
         self.refresh_view()
 
-        # Save current emoji filter to settings
-        s = get_settings()
-        setting_emoji_group = s.get('emoji_group_filter') or 'smileys-emotion'
-        current_emoji_group = get_app().window.emojiFilterGroup.currentData()
-        if setting_emoji_group != current_emoji_group:
-            s.set('emoji_group_filter', current_emoji_group)
-
-    def refresh_view(self, *args):
-        """Filter transitions with proxy class"""
-        filter_text = self.win.emojisFilter.text().strip().replace(' ', '.*')
-        current_emoji_group = get_app().window.emojiFilterGroup.currentData()
-        if current_emoji_group != "Show All":
-            # Create regex which combines category and filter text
-            # It's much faster for Qt to do this regex than use a custom
-            # Python filter function
-            # FYI:    smiley.*alien.*|alien.*smiley.*
-            filter_text = "%s.*%s.*|%s.*%s.*" % (filter_text, current_emoji_group, current_emoji_group, filter_text)
-        self.emojis_model.proxy_model.setFilterRegExp(QRegExp(filter_text.replace(' ', '.*')))
-        self.emojis_model.proxy_model.setFilterCaseSensitivity(Qt.CaseInsensitive)
-        self.emojis_model.proxy_model.sort(Qt.AscendingOrder)
+    def refresh_view(self):
+        col = self.model.sortColumn()
+        self.model.sort(col)
```

Comment:
In addition to cutting out all of that complexity, `self.emojis_model.proxy_model.sort(Qt.AscendingOrder)` was actually not working, since `QAbstractItemModel.sort()` takes a column index as its first argument. It's just pure luck that `Qt.AscendingOrder` happens to be the 0 item in the enum, so it was sorting column 0 instead of crashing.

Reply:
Nice catch!